src/pages/SupplierInvoicePage.js

```diff
@@ -1,100 +1,201 @@
-/* @flow weak */
-
 /**
  * mSupply Mobile
  * Sustainable Solutions (NZ) Ltd. 2016
  */
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import { View } from 'react-native';
-
-import { PageInfo } from '../widgets';
-import { formatDate, parsePositiveInteger, sortDataBy } from '../utilities';
+import autobind from 'react-autobind';
+import {
+  formatDate,
+  parsePositiveFloat,
+  parsePositiveInteger,
+  formatExpiryDate,
+  parseExpiryDate,
+  sortDataBy,
+} from '../utilities';
+import { createRecord } from '../database';
 import { GenericPage } from './GenericPage';
 import globalStyles from '../globalStyles';
-import { pageInfoStrings, tableStrings } from '../localization';
+import { buttonStrings, modalStrings, pageInfoStrings, tableStrings } from '../localization';
+import {
+  AutocompleteSelector,
+  BottomConfirmModal,
+  PageButton,
+  PageContentModal,
+  PageInfo,
+  TextEditor,
+} from '../widgets';
 
 const DATA_TYPES_SYNCHRONISED = ['TransactionItem', 'TransactionBatch', 'Item', 'ItemBatch'];
 
-export class SupplierInvoicePage extends GenericPage {
+const MODAL_KEYS = {
+  COMMENT_EDIT: 'commentEdit',
+  THEIR_REF_EDIT: 'theirRefEdit',
+  ITEM_SELECT: 'itemSelect',
+};
+
+const SORT_DATA_TYPES = {
+  itemName: 'string',
+  itemCode: 'string',
+  batch: 'string',
+  numberOfPacks: 'number',
+  packSize: 'number',
+};
+
+export class SupplierInvoicePage extends React.Component {
   constructor(props) {
     super(props);
-    this.state.sortBy = 'itemName';
-    this.state.columns = [
-      {
-        key: 'itemCode',
-        width: 1,
-        title: tableStrings.item_code,
-        sortable: true,
-      },
-      {
-        key: 'itemName',
-        width: 2,
-        title: tableStrings.item_name,
-        sortable: true,
-      },
-      {
-        key: 'totalQuantitySent',
-        width: 1,
-        title: tableStrings.number_sent,
-        sortable: true,
-        alignText: 'right',
-      },
-      {
-        key: 'numReceived',
-        width: 1,
-        title: tableStrings.number_received,
-        sortable: true,
-        alignText: 'right',
-      },
-    ];
-    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
-    this.finalisableDataType = 'Transaction';
-    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
-    this.onEndEditing = this.onEndEditing.bind(this);
-    this.onDatabaseEvent = this.onDatabaseEvent.bind(this);
+    this.dataFilters = {
+      searchKey: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    this.state = {
+      totalPrice: 0,
+      modalKey: null,
+      modalIsOpen: false,
+      selection: [],
+    };
+    autobind(this);
+  }
+
+  updateDataFilters(newSearchTerm, newSortBy, newIsAscending) {
```

Comment:
my eslint complains here on updateDataFilters line: `updateDataFilters should be placed after getModalTitle (react/sort-comp)`